S2Geometry/DataStructures/NullObject.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Google.Common.Geometry
+{
+    internal struct NullObject<T> : IEquatable<T>
+    {
+        [DefaultValue(false)]
+        private bool isNotNull;// default property initializers are not supported for structs
+
+        private NullObject(T item, bool isnull) : this()
+        {
+            this.isNotNull = isNotNull;
```

Comment:
this should be `this.isNotNull = !isNull` if you're keeping the ctor as `isnull`